### PR TITLE
Fixed the CSS of Eli Selkin card in Engage's Project page was: displayed nonuniformly

### LINKED ISSUE #7429: Card text for Eli Selkin card in Engage's Project page is: displayed nonuniformly
### Overview
In the Engage project page, the text for Eli Selkin's card does not have the same visual margin as the other cards when displayed in phone portrait view. 

### Action Items
- [x] Replicate the issue by using the inspect tool in chrome and looking at the emulator for a mobile phone.
- [x]  Check other project pages to see if it's happening on their current team member listings
- [x]  If it is, review code in resource 2 and see if it has a problem.
- [x] if it is not, review md file for the project (resource 3) and see if there is an anomaly there.
- [x] Write up the problem, and the proposed solution in the comments on the issue
- [x]  Review with merge team member (during office hours or a team meeting) if you are unsure of anything about it
- [x]  Implement solution, and test locally
- [x] Put in PR.

### Resources/Instructions

1. The engage project page: https://www.hackforla.org/projects/engage

2. The HTML that controls the layout of the project pages: https://github.com/hackforla/website/blob/gh-pages/_layouts/project.html

3. MD file for the project https://github.com/hackforla/website/blob/gh-pages/_projects/engage.md

4. The screenshot that shows the anomaly
<details><summary>screenshot</summary>

</details>

### PR DESCRIPTION
**Pull request by codyyjxn**
Fixes #7429

### What changes did you make?
  
  - I resolved the issue by adjusting the `@media #{$bp-below-mobile}`  to `flex-direction: row;` , ensuring it didn't impact the landscape view or responsiveness across mobile and web platforms. I also verified that the changes-maintained user-friendliness and dynamic functionality throughout.
 - Also added  `flex: 1`  to the description for each card. This gave a seamless interaction for the user.
  
### Why did you make the changes (we will use this info to test)?
  
- The reason I made these changes in engage project page, was because the text for Eli Selkin's card did not 
  have the same visual margin as the other cards when displayed in phone portrait view.
  - I addressed this issue to ensure it aligns with the user-friendly standards we uphold at HackForLA

### Screenshots of Proposed Changes To The Website (if any, please do not include screenshots of code changes)
 

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
newscreen

</details>

### CODE REVIEW

**Comment by tamara-snyder** [comment]
**Review ETA:** End of day 9/22/2024
**Availability:** Morning and evening

File: CNAME [hunk 1]
```diff
@@ -1 +1 @@
-www.hackforla.org
\ No newline at end of file
+www.hackforla.org
```
Comment (tamara-snyder) [inline CNAME]: Did anything change on this line, or is it just GitHub being silly? I don't think this affects anything so I'm fine with it

**Review by tamara-snyder — COMMENTED** [review]
Hi Jonathan, I think the padding solution you've come up with works well. I noticed the same issue was happening in the following places, and it looks like your fix has resolved the issues in those spots as well 😄 

- 311 Data:
  - Anna Kim
- Heart:
  - Charlotte Soestini
- Expunge Assist:
  - Maria Weissman
- LA TDM Calculator:
  - Bonnie Wolfe
- Tech Work Experience:
  - Joshua Fishman
- EMS Triage Tracker:
  - Shawn Duenas
  - Simone Wojtaszek
  - Nicole Doan
  - Conor McKiernan
  - Selena Jiang
  - Kelvin Nguyen
- Civic Tech Index:
  - Bonnie Wolfe
  - Bhaggyalakshmi Balasubramaniyan
  - Maxwell Countryman Skewes

I did notice that a bit of text is spilling off the cards at around 487x1183. Is it feasible/in scope to resolve this within this issue, perhaps by moving the text to the left a bit?
Screenshot 2024-09-22 100803

Otherwise, looks great! Really nice work on this!

File: CNAME (hunk 1 above)
Comment (codyyjxn) [reply CNAME]: I’ve been trying to delete this particular change, but it keeps reappearing every time I attempt to remove it. I'm not sure why this is happening, and I would appreciate any advice on how to fix this issue.

**Comment by santiseccovidal** [comment]
Review ETA: 6 PM 9/23/2024
Availability: 1-3 PM Monday-Tuesday

File: CNAME (hunk 1 above)
Comment (santiseccovidal) [reply CNAME]: I think there was an extra tab probably or something. But it's not important.

**Comment by daras-cu** [comment]
@codyyjxn as it appears the display issue is only occurring at mobile size, you really only need the increased padding under the mobile breakpoint (`@media #{$bp-below-mobile}`). This should resolve the issue @tamara-snyder noted with the text being off to the right/overflowing at larger screen sizes.

**Comment by codyyjxn** [comment]
@tamara-snyder @daras-cu Thank you for the feedback I went ahead and implemented it seems to work better. Thanks again! Can you please review Thanks !

**Review by santiseccovidal — DISMISSED** [review]
Hi @codyyjxn great job! Together with the last changes, suggested by @daras-cu, the view improves considerably.
Some text, similar to what @tamara-snyder noted, still spills off around 484×5363.
However, I don't know if that can be further improved.
Other than that, the code is correct, as well as your explanation of your changes.
Thanks for working on this issue.

**Comment by codyyjxn** [comment]
> Hi @codyyjxn great job! Together with the last changes, suggested by @daras-cu, the view improves considerably. Some text, similar to what @tamara-snyder noted, still spills off around 484×5363. However, I don't know if that can be further improved. Other than that, the code is correct, as well as your explanation of your changes. Thanks for working on this issue.

@santisecco Oh! Could you please share a screenshot of what you're seeing on your end, particularly where the issue is still occurring? I’d like to take a look and see how I can improve it, if necessary.

File: _sass/components/_project-page.scss [hunk 2]
```diff
@@ -410,7 +410,6 @@
 }
 
 .leader-description {
-  padding: 0px 12px;
```
Comment (daras-cu) [inline _sass/components/_project-page.scss]: @codyyjxn this line needs to be added back in now that you moved the 60px padding down to the mobile device section. Then you should be good.

**Comment by santiseccovidal** [comment]
Hi @codyyjxn I couldn't replicate the problem still, now that I think about it, I believe the dimension I mentioned 484x5363 is not usual. Honestly my exp in front end is limited. So anyways, using other dimensions there are still some issues I hadn't seen before:
Screenshot 2024-09-24 121721
Screenshot 2024-09-24 121804
Screenshot 2024-09-24 121922

### Though here's one that looks good 👍

Screenshot 2024-09-24 121945

**Comment by codyyjxn** [comment]
@tamara-snyder @santisecco @daras-cu 
Thank you all for your input and I have committed all the changes now. I'm ready for a review. Thank you again!

**Review by daras-cu — DISMISSED** [review]
@codyyjxn thanks for making those changes. The text displays nicely at all sizes now, great job.

**Comment by santiseccovidal** [comment]
Hey @codyyjxn now it looks much better.
Screenshot 2024-09-25 153120
Screenshot 2024-09-25 153140
However, I'd like first to hear what @tamara-snyder thinks about it. I believe that the phone look still looked better with the other changes recommended by her. I'll try to check out a solution to that

**Comment by santiseccovidal** [comment]
Ok so I've been trying to use 
`
@media (max-width: 432px) {
}
`
(I also saw as ok, perhaps more "declarative")
`
@media screen and (max-width: 432px) {
}
`
To apply the changes in the case that the width of the screen is wide until 432px, obviously 432px or a bit larger as you find suitable.
I don't know if this is a good practice, is too complex, etc.
But I thought that if we can specify a width and change it there so that the names align better from the left, then it could be better/nicer to the eye.

Just let me know what you think @codyyjxn

**Comment by codyyjxn** [comment]
@tamara-snyder @santisecco 
Thank you for your valuable input. I’ve implemented the suggested changes where applicable. When you have a moment, could you review my pull request? I appreciate your time and feedback.

**Review by tamara-snyder — DISMISSED** [review]
Hey @codyyjxn, thanks for your patience and for trying out all these changes! I agree with @santisecco here in that I've been hoping to find a better way to line up the names from the left. I was playing with this and found that on line 419, setting the `min-width: 100%` (in the .leader-description bp-below-mobile section) aligns the names nicely. It has to be `min-width: 100%` for whatever reason... setting `width: 100%` did not work for me. 

When you have a chance, could you please try that out as a potential solution?

P.S. I only see two lines changed in your current PR. Are there more that were meant to be included as part of your latest update?

Screenshot 2024-09-26 072705

**Comment by tamara-snyder** [comment]
One more thing I just thought of! Once this is finished, it would probably be a good idea to coordinate these changes with the design team since the .leader-description section of the [Figma design page](https://www.figma.com/design/0RRPy1Ph7HafI3qOITg0Mr/Hack-for-LA-Website?node-id=3464-3&node-type=canvas&t=HxpICqojf2JhSQTX-0) doesn't yet include a bp-below-mobile section.

**Comment by 8alpreet** [comment]
Hi everyone!

I played around with this code and here are a couple things I learned.

#### The current approach:
* fundamentally, we're fighting the `align-items: center` located in the media query under `.leader-card` class. The whole element is centered, but the text is not because it is of varying sizes. This problem becomes apparent when one of the description fields becomes large.
* So we have to do something about that alignment in order for the leader-description elements to be left-aligned across all the cards.
* My solution to this is to use `align-self: flex-start`
* Now, description elements are aligned but they're too far to the left. So we need some margin, for which I added `margin: 0 10% 0 20%;`
* Through this experimentation, I found that almost all the styling currently applied within the media-query under `.leader-description` class it not necessary. Making it a display flex and aligning items doesn't make sense because we have no issue with the alignment of the `leader-description-field` elements (which are the children). The issue is with the parent element.
* work with these changes locally: 
    ```
    git checkout -b 8alpreet-alternate-styling-v2-7429
    git pull https://github.com/8alpreet/website.git alternate-styling-v2-7429
    ```

#### An alternate approach:
* So if we don't want to fight the `align-items:center` property. We can take a different approach.
* We can keep modify the media query within the `.leader-card` class to use `flex-direction: row`. 
* Then we can control the growth of the description element by using `flex: 1`
* The effect of this looks better in my opinion and simplifies the code.
* This works with very long entries in the description field as well. Checkout Bonnie's role in projects/tdm-calculator
* work with these changes locally:
  ```
   git checkout -b 8alpreet-alternate-styling-7429
   git pull https://github.com/8alpreet/website.git alternate-styling-7429
  ```

I played around with this for a few hours, but I'm open to further collaboration to explore this further.

**Comment by codyyjxn** [comment]
> Hi everyone!
> 
> I played around with this code and here are a couple things I learned.
> 
> #### The current approach:
> * fundamentally, we're fighting the `align-items: center` located in the media query under `.leader-card` class. The whole element is centered, but the text is not because it is of varying sizes. This problem becomes apparent when one of the description fields becomes large.
> * So we have to do something about that alignment in order for the leader-description elements to be left-aligned across all the cards.
> * My solution to this is to use `align-self: flex-start`
> * Now, description elements are aligned but they're too far to the left. So we need some margin, for which I added `margin: 0 10% 0 20%;`
> * Through this experimentation, I found that almost all the styling currently applied within the media-query under `.leader-description` class it not necessary. Making it a display flex and aligning items doesn't make sense because we have no issue with the alignment of the `leader-description-field` elements (which are the children). The issue is with the parent element.
> * work with these changes locally:
>   ```
>   git checkout -b 8alpreet-alternate-styling-v2-7429
>   git pull https://github.com/8alpreet/website.git alternate-styling-v2-7429
>   ```
> 
> #### An alternate approach:
> * So if we don't want to fight the `align-items:center` property. We can take a different approach.
> * We can keep modify the media query within the `.leader-card` class to use `flex-direction: row`.
> * Then we can control the growth of the description element by using `flex: 1`
> * The effect of this looks better in my opinion and simplifies the code.
> * This works with very long entries in the description field as well. Checkout Bonnie's role in projects/tdm-calculator
> * work with these changes locally:
>   ```
>    git checkout -b 8alpreet-alternate-styling-7429
>    git pull https://github.com/8alpreet/website.git alternate-styling-7429
>   ```
> 
> I played around with this for a few hours, but I'm open to further collaboration to explore this further.

Hey @8alpreet  Thank you so much for your recommendations! After experimenting with the various implementations, I found that the simpler code really resonates with me it looks clean and easy to follow. Great work, and I appreciate your help! I will be implanting the code right now.

**Comment by codyyjxn** [comment]
Thank you all for your help and guidance. Please review my pr when possible. 
@tamara-snyder @santisecco @daras-cu @8alpreet

**Comment by santiseccovidal** [comment]
I really like the way it looks now. Great job guys. Thanks for the tips and explanation @8alpreet

**Comment by santiseccovidal** [comment]
However, I just noticed that following the way we've been working, together with @tamara-snyder suggestions, the picture is above the text (name, role). For some reason I like that more for the mobile version.

**Comment by 8alpreet** [comment]
@codyyjxn Thanks for your continued work on this! Whichever design we settle upon, could you please make sure to update the PR comment.

**Comment by santiseccovidal** [comment]
Did you try out her changes @codyyjxn? It would be good to have both version, so that then we can decide which one to choose

**Comment by codyyjxn** [comment]
Thank you all for your input. I have updated the PR comment based on the feedback. During office hours yesterday, I was advised to make the necessary changes from the original issue, and the merge team will notify me if any further adjustments are needed. They recommended that I seek approval and wait for the merge team’s decision.
@tamara-snyder @santisecco @daras-cu

**Comment by santiseccovidal** [comment]
Thanks for this conversation, it really helped me understand front-end a lot more.
@codyyjxn understood, please only notice that currently the mobile version of the Engage project looks like this:
Screenshot 2024-09-27 150002

With the last changes, not only #7429 Eli Selkin's card (as stated in the issue) will be modified but all the other cards also. Resulting in this view (I'm posting the screenshots so that other developers can see)

Screenshot 2024-09-27 154021

I was not present in the meeting yesterday but will attend Sunday's meeting. I believe that the issue looked for a different solution rather than modifying all the cards which I believe would go outside the scope of issue #7429 in my opinion.

@codyyjxn I understand that there were many suggestions made.

**Comment by 8alpreet** [comment]
@santisecco 

This problem is not localized to a single project and it cannot be because we're using the same underlying ui-elements across all projects.

I found the problem in the following:
1. `/projects/311-data`
2. `/projects/civic-tech-jobs`
3. `/projects/expunge-assist`

The problem will occur in every project with long entires in the description fields. The current styling is fragile.

**Comment by santiseccovidal** [comment]
@8alpreet yes I had noticed that the changes affect all cards. 
Great check noticing and listing the problems on each project.

I agree that changing the layout will solve all those issues. It looks great also.
I would like to check the changes in my phone to see the difference. I just did and the perception of the website changes interestingly.

**Comment by santiseccovidal** [comment]
Ok just to give an update @tamara-snyder @codyyjxn @8alpreet @daras-cu. We'll be following @8alpreet solution. So the last PR version is the one to check.
It was decided with Bonnie @tamara-snyder and the rest of devs in today's meeting.

**Comment by codyyjxn** [comment]
> Ok just to give an update @tamara-snyder @codyyjxn @8alpreet @daras-cu. We'll be following @8alpreet solution. So the last PR version is the one to check. It was decided with Bonnie @tamara-snyder and the rest of devs in today's meeting.

Thank you all for the update I wasn't able to attend the meeting today but I really got to understand the UI a lot better. Thanks all for the input.  Is there anything I need to improve before this gets merged ? @tamara-snyder @santisecco @8alpreet @daras-cu

**Comment by 8alpreet** [comment]
@codyyjxn, thanks for the update! I will complete my review 7pm today.

**Review by 8alpreet — DISMISSED** [review]
Hi @codyyjxn,

Great job on this issue, I know it became complex and took longer than we expected. You stuck through it, so props for that! I locally tested the changes across different projects and the cards look great.

#### Things you did well:
* PR title is descriptive
* Issue is Linked
* Branch name includes issue number
* What section describes the changes
* Before and after pictures clearly outline the changes

#### Room for Improvement:
* OPINION - The PR title can be updated since the current one, though accurate to the issue, does not accurately reflect the changes. So something like "Updated CSS to fix text alignment issue in project leader card" could work.
* If you want the keep the current title, the "--7429" portion is definitely not necessary.
* OPINION - The why section could use more specific details; although you answer the question, it is high level and a bit vague. A quick insight into why these **specific** changes were needed would be wonderful.

#### Notes:
* the CNAME file should be not changed but looks like that issue was already discussed and resolved.

**Comment by codyyjxn** [comment]
@tamara-snyder @santisecco @daras-cu  Please review when possible. Thank you !

**Review by daras-cu — DISMISSED** [review]
Thanks for continuing to work on this @codyyjxn, I think having the text remain next to the picture for smaller screen sizes is nicer visually and more consistent with how the page displays at larger sizes. I know it went a little beyond the scope of the original issue, but I'm glad you got to learn more about the UI and page styling in the process. 

Everything looks good when I test your branch locally and your PR is set up correctly. Well done!

**Review by santiseccovidal — DISMISSED** [review]
Hi @codyyjxn thanks for working on this issue. As discussed in the meeting the last layout is the one to be adopted and it's looking great. Also the approach chosen mentioned by @8alpreet simplifies the code in my opinion.

I would like to also thank the code suggestions and help of @daras-cu, @8alpreet and @tamara-snyder. They for sure contributed a big deal to solving this.

Great job

**Review by tamara-snyder — DISMISSED** [review]
Hey @codyyjxn, great work on this and thanks for all your patience trying out various changes! I think you've executed this approach well. Thanks also to @8alpreet, @santisecco, and @daras-cu for sharing your thoughts and solutions - I learned some useful tips from this discussion.

**Comment by jphamtv** [comment]
Hey @codyyjxn, great work collaborating with everyone on this PR to come up with an elegant solution. 

This is ready to be merged, but first, can you please fix the CNAME file? I noticed there's an extra blank line (line 2) introduced, which should be removed. Having extra blank lines in a CNAME file may cause issues with DNS resolution, potentially affecting the domain configuration.

Screenshot 2024-10-03 at 16 22 21


Thanks!

**Comment by codyyjxn** [comment]
> Hey @codyyjxn, great work collaborating with everyone on this PR to come up with an elegant solution.
> 
> This is ready to be merged, but first, can you please fix the CNAME file? I noticed there's an extra blank line (line 2) introduced, which should be removed. Having extra blank lines in a CNAME file may cause issues with DNS resolution, potentially affecting the domain configuration.
> 
> Screenshot 2024-10-03 at 16 22 21
> Thanks!

Hello @jphamtv . I tried to uncommit the CNAME file but for some reason I can't seem to get rid of that change. What would you recommend I do ?

**Comment by jphamtv** [comment]
> Hello @jphamtv . I tried to uncommit the CNAME file but for some reason I can't seem to get rid of that change. What would you recommend I do ?

@codyyjxn - Thanks for trying to resolve this. Since uncommitting didn't work, let's do it manually:

1. Open the CNAME file in your editor
2. Delete the blank line (line 2)
3. Commit and push the changes

**Comment by codyyjxn** [comment]
> > Hello @jphamtv . I tried to uncommit the CNAME file but for some reason I can't seem to get rid of that change. What would you recommend I do ?
> 
> @codyyjxn - Thanks for trying to resolve this. Since uncommitting didn't work, let's do it manually:
> 
> 1. Open the CNAME file in your editor
> 2. Delete the blank line (line 2)
> 3. Commit and push the changes

Thanks ! This worked I was able to remove that file from this pr. @jphamtv 
Do I need to request another review for this pr?

**Review by 8alpreet — APPROVED** [review]

**Review by jphamtv — APPROVED** [review]
Thanks for the quick update @codyyjxn, approved and merged.

**Pull request merged**